hathor/transaction/token_creation_tx.py

```
@@ -65,16 +62,20 @@ def __init__(
             storage=storage,
             settings=settings,
         )
+        self.token_info_version = (
+            TokenInfoVersion.DEPOSIT if self._settings.FEE_FEATURE_FLAG is False else token_info_version)
```

I would let txs be created normally with whatever token info version is passed to them. It's the verifier's job to prevent disabled versions. It's better to let them fail instead of silently changing the version.

---

Done. I've moved this validation to `token_creation_transaction_verifier.py`

---

Thanks!